src/views/templates/actions/hooks/useVirtualMachineTemplatesActions.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+
+import { TemplateModel, V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import DeleteModal from '@kubevirt-utils/components/DeleteModal/DeleteModal';
+import { useModal } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Action, k8sDelete } from '@openshift-console/dynamic-plugin-sdk';
+
+import CloneTemplateModal from '../components/CloneTemplateModal';
+
+type useVirtualMachineTemplatesActionsProps = (template: V1Template) => Action[];
+
+const useVirtualMachineTemplatesActions: useVirtualMachineTemplatesActionsProps = (
+  template: V1Template,
+) => {
+  const { t } = useKubevirtTranslation();
+  const isCommonTemplate = template.metadata.labels['template.kubevirt.io/type'] === 'base';
+  const { createModal } = useModal();
+
+  const actions = [
+    {
+      id: 'edit-template',
+      label: t('Edit Template'),
+      cta: () => console.log('Edit Template'), // TODO lead to the template details page
+    },
+    {
+      id: 'clone-template',
+      label: t('Clone Template'),
+      cta: () =>
+        createModal(({ isOpen, onClose }) => (
+          <CloneTemplateModal obj={template} isOpen={isOpen} onClose={onClose} />
+        )),
+    },
+    {
+      id: 'edit-boot-source',
+      label: t('Edit boot source'),
+      disabled: isCommonTemplate, // common templates cannot be edited or deleted
+      cta: () => console.log('Edit boot source'),
+      // TODO add the modal
+      // cta: () =>
+      //   createModal(({ isOpen, onClose }) => (
+      //     <EditBootSourceModal obj={template} isOpen={isOpen} onClose={onClose} />
+      //   )),
+    },
+    {
+      id: 'delete-template',
+      label: t('Delete Template'),
+      disabled: isCommonTemplate,
+      cta: () =>
+        createModal(({ isOpen, onClose }) => (
+          <DeleteModal
+            obj={template}
+            isOpen={isOpen}
+            onClose={onClose}
+            headerText={t('Delete Virtual Machine Template?')}
+            onDeleteSubmit={() =>
+              k8sDelete({
+                model: TemplateModel,
+                resource: template,
+                json: undefined,
```

Comment:
this is very strange, why do you need `json` and `requestInit` to delete the template here ?